### PR TITLE
Add struct extension

### PR DESCRIPTION
**Pull request by flash-gordon**
It simply infers a hash schema from a struct. More complex scenarios are not supported, there are reasons for this. In future, we plan to infer structs from schema definition, similarly to how it works in rom.

### CODE REVIEW

**Comment by solnic** [comment]
> More complex scenarios are not supported

@flash-gordon could you give a couple of examples? 🤔

File: lib/dry/schema/extensions/struct.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+require 'dry/schema/macros/hash'
+
+module Dry
+  module Schema
+    module Macros
+      Hash.prepend(::Module.new {
+        def call(*args)
+          if args.size >= 1 && args[0].is_a?(::Class) && args[0] <= ::Dry::Struct
+            if block_given?
+              raise ArgumentError, 'blocks are not supported'
```
Comment (solnic) [inline lib/dry/schema/extensions/struct.rb]:
```suggestion
              raise ArgumentError, "blocks are not supported when using a struct class (#{name} => #{args[0]})"
```

**Comment by flash-gordon** [comment]
@solnic it's something insane like `value(StructOne, StructTwo)` or 
```ruby
required(:foo).value(User) do
  require(:i_have_no_idea_what_im_doing).value(:string)
end
```

**Comment by solnic** [comment]
@flash-gordon 😆 then we're fine

**Pull request merged**